### PR TITLE
`Command` and `Event` validation in factory

### PR DESCRIPTION
**Pull request by armiol**
This PR enables a mandatory validation of `Event` and `Command` instances created in `EventFactory` and `CommandFactory` respectively. Each message passed is validated according to its Protobuf definition.

In case an invalid `Message` instance is passed, a `ConstraintViolationThrowable` is thrown.

Also, `CommandFactory` is now a single point that allows command creation. `Commands` utility no longer provides an API  for command and command context creation.

The framework version is set to `0.9.13-SNAPSHOT`.

### CODE REVIEW

**Comment by armiol** [comment]
@alexander-yevsyukov PTAL.

File: client/src/main/java/org/spine3/client/CommandFactory.java [hunk 1]
```diff
@@ -20,17 +20,39 @@
...
  * configuration.
  *
+ * <p>During the creation of {@code Command} instances the source {@code Message} instances, passed
+ * into creation methods, are validated. The validation is performed according to the constraints
+ * set in Protobuf definition of each {@code Message}. In case the message isn't valid,
+ * an {@linkplain ConstraintViolationThrowable exception} is thrown.
+ *
+ * <p>Therefore it is recommended to use a corresponding
+ * {@linkplain org.spine3.validate.ValidatingBuilder validating Builder} implementation to create
+ * the Command message.
```
Comment (alexander-yevsyukov) [inline client/src/main/java/org/spine3/client/CommandFactory.java]: I'd say `a command message.`

File: client/src/main/java/org/spine3/client/CommandFactory.java [hunk 2]
```diff
@@ -20,17 +20,39 @@
...
  * Public API for creating {@link Command} instances, using the {@code ActorRequestFactory}
  * configuration.
  *
+ * <p>During the creation of {@code Command} instances the source {@code Message} instances, passed
+ * into creation methods, are validated. The validation is performed according to the constraints
+ * set in Protobuf definition of each {@code Message}. In case the message isn't valid,
+ * an {@linkplain ConstraintViolationThrowable exception} is thrown.
+ *
+ * <p>Therefore it is recommended to use a corresponding
+ * {@linkplain org.spine3.validate.ValidatingBuilder validating Builder} implementation to create
```
Comment (alexander-yevsyukov) [inline client/src/main/java/org/spine3/client/CommandFactory.java]: Why not `ValidatingBuilder`? After all we refer to particular kind of builders that we generate.

File: client/src/main/java/org/spine3/client/CommandFactory.java [hunk 3]
```diff
@@ -48,48 +70,210 @@
...
-                                      actorRequestFactory.getZoneOffset());
-    }
-
-    /**
      * Creates new {@code Command} with the passed message and target entity version.
      *
      * <p>The command contains a {@code CommandContext} instance with the current time.
      *
+     * <p>The message passed is validated according to the constraints set in its Protobuf
+     * definition. In case the message isn't valid, an {@linkplain ConstraintViolationThrowable
```
Comment (alexander-yevsyukov) [inline client/src/main/java/org/spine3/client/CommandFactory.java]: What we we rename  `ConstraintViolationThrowable` to `ValidationException`?

Comment (armiol) [inline client/src/main/java/org/spine3/client/CommandFactory.java]: As discussed vocally, there is an issue #482. I have added the `ValidationException` as another suggestion for renaming.

**Comment by armiol** [comment]
@alexander-yevsyukov the issues were addressed. PTAL.

**Pull request merged**